File: 3309-count-prefix-and-suffix-pairs-i/count-prefix-and-suffix-pairs-i.cpp
class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {
        int ans = 0;
        for (int i=0; i<words.size()-1;i++){
            for(int j=i+1; j<words.size();j++){
                if(isPrefixAndSuffix(words[i],words[j])){
                    ans++;
                }
            }
        }
        return ans;
    }

    bool isPrefixAndSuffix(string str1, string str2){
        if(str2.size()<str1.size()){
            return false;
        }
        string p = str2.substr(0,str1.size());
        string s = str2.substr(str2.size()-str1.size(),str2.size());
        // cout<<str1<<" "<<str2<<endl;
        // cout<<"p "<<p<<" s "<<s<<endl;


        if(p==s && s==str1){
            return true;
        }
        return false;

    }
};

File: 3309-count-prefix-and-suffix-pairs-i/count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        ans = 0
        for i in range(len(words)-1):
            for j in range(i+1,len(words)):
                if self.isPrefixAndSuffix(words[i],words[j]):
                    ans+=1
        return ans
    

    def isPrefixAndSuffix(self,str1,str2):
        if len(str2)<len(str1):
            return False
        p = str2[:len(str1)]
        s = str2[-len(str1):]
        if str1==p and str1==s:
            return True
        return False


        